File: 上机1/pro2.py
from decimal import *
from turtle import end_fill
x = Decimal('100000')
# getcontext().prec = 100
sum = -Decimal.ln(x+Decimal.sqrt(x*x-Decimal('1')))
print(sum)
result = [Decimal('-12.2060728'), Decimal('-12.206073762186564'), Decimal(
    '-12.206072645505174')]
error = [0, 0, 0]
for i in range(3):
    error[i] = abs((result[i]-sum)/sum)
print(sum)
print(error)

File: 上机1/pro2.cpp
#include <iostream>
#include <math.h>
#include <iomanip>
#include "stdlib.h"
using namespace std;
int main()
{
    long double x = 100000;
    long double sum;
    sum = -log(x + sqrt(x * x - 1));
    sum = -sum;
    long double result[3] = {12.2060728,
                             12.206073762186564,
                             12.206072645505174};
    long double error[3];
    cout << setprecision(100) << sum << endl;
    cout << setprecision(100) << result[2] << endl;
    for (int i = 0; i < 3; i++)
    {
        error[i] = abs((result[i] - sum) / sum);
        cout << setprecision(100) << error[i] << endl;
    }
}